### PR TITLE
Make build mac os test hermetic

### PR DESCRIPTION
**Pull request by Jasguerrero**
Part of: #107607

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter_tools/test/commands.shard/hermetic/build_linux_test.dart [hunk 1]
```diff
@@ -611,6 +643,8 @@ set(BINARY_NAME "fizz_bar")
 
   testUsingContext('Performs code size analysis and sends analytics', () async {
     final BuildCommand command = BuildCommand(
+      flutterUsage: TestUsage(),
```
Comment (christopherfujino) [inline packages/flutter_tools/test/commands.shard/hermetic/build_linux_test.dart:646]: can Usage and Platform overrides be removed from these tests?

Comment (christopherfujino) [reply packages/flutter_tools/test/commands.shard/hermetic/build_linux_test.dart:646]: and if not, maybe we should be leaving TODO comments on what needs to happen before they can be removed.

File: packages/flutter_tools/test/commands.shard/hermetic/build_macos_test.dart [hunk 2]
```diff
@@ -175,14 +177,14 @@ STDERR STUFF
       const <String>['build', 'macos', '--no-pub']
     ), throwsA(isA<UsageException>()));
   }, overrides: <Type, Generator>{
-    Platform: () => notMacosPlatform,
```
Comment (christopherfujino) [inline packages/flutter_tools/test/commands.shard/hermetic/build_macos_test.dart:178]: nice!

File: packages/flutter_tools/test/commands.shard/hermetic/build_macos_test.dart [hunk 3]
```diff
@@ -175,14 +177,14 @@ STDERR STUFF
...
     FeatureFlags: () => TestFeatureFlags(isMacOSEnabled: true),
   });
 
   testUsingContext('macOS build fails when feature is disabled', () async {
     final BuildCommand command = BuildCommand(
+      flutterUsage: TestUsage(),
+      platform: macosPlatform,
       androidSdk: FakeAndroidSdk(),
       buildSystem: TestBuildSystem.all(BuildResult(success: true)),
       fileSystem: MemoryFileSystem.test(),
```
Comment (christopherfujino) [inline packages/flutter_tools/test/commands.shard/hermetic/build_macos_test.dart:190]: looks like we're mixing file systems here. Also, is anything grabbing the filesystem from the context override?

File: packages/flutter_tools/test/commands.shard/hermetic/build_linux_test.dart (hunk 1 above)
Comment (Jasguerrero) [reply packages/flutter_tools/test/commands.shard/hermetic/build_linux_test.dart:646]: The `BuildLinuxCommand` should be migrated first to delete those, this PR is just for `BuildMacosCommand`

File: packages/flutter_tools/test/commands.shard/hermetic/build_macos_test.dart (hunk 3 above)
Comment (Jasguerrero) [reply packages/flutter_tools/test/commands.shard/hermetic/build_macos_test.dart:190]: We still need fileSystem override is used on `flutter_command.dart` on `verifyThenRunCommand`

File: packages/flutter_tools/test/commands.shard/hermetic/build_linux_test.dart (hunk 1 above)
Comment (christopherfujino) [reply packages/flutter_tools/test/commands.shard/hermetic/build_linux_test.dart:646]: ahhh, good point

File: packages/flutter_tools/test/commands.shard/hermetic/build_macos_test.dart (hunk 3 above)
Comment (christopherfujino) [reply packages/flutter_tools/test/commands.shard/hermetic/build_macos_test.dart:190]: then you should pass the same one we're passing to the BuildCommand. Each test should have at most one MemoryFileSystem.

**Comment by christopherfujino** [comment]
@Jasguerrero take a look at Windows build_tests_3_3

**Pull request closed without merging**